Documentation/General/emergentArchitectureDesign/emergentArchitectureDesign.tex

```diff
@@ -191,3 +191,47 @@ \section{Concurrency}
 These seperate threads will get a specific task assigned, which has his local data to compute on.
 Deadlocks will be prevented because the spawned threads will be joined before acquiring locks in synchronized statements.
 This will happen when we persist the algorithm output in the database.
+
+\section{Design Patterns}
+
+Due to the usage of various libraries and frameworks, our codebase does not contain a lot of helper mechanisms.
+Therefore the number of design patterns implemented is lower, because we use a lot of patterns defined in the frameworks.
+
+The design patterns we use, as they are defined in the frameworks, are:
+
+\begin{enumerate}
+\item Lazy initialization (Hibernate):\\
+OneToMany-relations are fetched from the database when they are needed.
+This makes retrieval of entities that use embeddables faster. For example we are most of the time not interested in the \class{VAVector} of a \class{Room}, only when we need to update such a \class{Room}, which happens rarely.
+
+\item Object pooling (RestEasy, Guice):\\
+The http-connection to SoundCloud and database-connection to our PostgreSQL database are pooled for re-use.
+This way we only have to set up the connection once using said frameworks and reduce object retrieval time.
+The \class{DBModule} uses a JpaPersistModule which is injected by Guice, whereas the \class{SoundCloudAPIConnector} uses a RestEasyClient.
```

Comment:
Almost true. The `JpaPersistModule` enables Guice to automatically initialize the `EntityManger` in the correct persistence context (`persistence.xml`). This has nothing to do with connection pooling. The connection pooling is done for database connections (in our case, either the Postgres JDBC or H2 driver). The pooling is done by the C3P0 (not kidding) extension for Hibernate.